Rewrite HTTP OAuth2 client BBEs (#3793)
ballerina-platform/ballerina-distribution: examples/http-client-oauth2-jwt-bearer-grant-type/http_client_oauth2_jwt_bearer_grant_type.md

# HTTP client - OAuth2 JWT bearer grant type

> The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration.

Author: ```suggestion
The `http:Client` can connect to a service that is secured with OAuth2 JWT bearer grant type authorization by enriching each request with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration.
```

> The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client wants to receive access tokens without transmitting sensitive information such as the client secret.

Reviewer: Lets remove this part `Use this grant type when the client wants to receive access tokens without transmitting sensitive information such as the client secret`

Author: Fixed via https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/373fd382554d45179c4ec44e8ea2280b4b8709b2

> The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client wants to receive access tokens without transmitting sensitive information such as the client secret.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration.
```

> An `http:Client` can be secured with OAuth2 JWT bearer grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 JWT bearer grant type can be specified in the `auth` field as the `http:OAuth2JwtBearerGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer 2: ```suggestion
The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 JWT bearer grant type can be specified in the `auth` field as the `http:OAuth2JwtBearerGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.
```

> An `http:Client` can be secured with OAuth2 JWT bearer grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 JWT bearer grant type can be specified in the `auth` field as the `http:OAuth2JwtBearerGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer 3: **When to use ?**
use when the client wants to receive access tokens without transmitting sensitive information such as the client secret

Author: Added via [187d99d](https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/187d99dffcd52f92e019e05f224ac4dc5e4a2e66)

> The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 JWT bearer grant type can be specified in the `auth` field as the `http:OAuth2JwtBearerGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 JWT bearer grant type can be specified in the `auth` field as the `http:OAuth2JwtBearerGrantConfig` record. Use this grant type when the client wants to receive access tokens without transmitting sensitive information such as the client secret.
```

> The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 JWT bearer grant type can be specified in the `auth` field as the `http:OAuth2JwtBearerGrantConfig` record. Use this grant type when the client wants to receive access tokens without transmitting sensitive information such as the client secret.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 JWT bearer grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client wants to receive access tokens without transmitting sensitive information such as the client secret.
```
